include/swift/Threading/Once.h

```diff
@@ -0,0 +1,36 @@
+//===--- Once.h - Runtime support for lazy initialization -------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// Swift runtime functions in support of lazy initialization.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_THREADING_ONCE_H
+#define SWIFT_THREADING_ONCE_H
+
+#include "Impl.h"
+
+namespace swift {
+
+using once_t = threading_impl::once_t;
+
+/// Runs the given function with the given context argument exactly once.
+/// The predicate argument must refer to a global or static variable of static
+/// extent of type swift::once_t.
+inline void once(once_t &predicate, void (*fn)(void *),
```

Comment:
May I suggest adding a templatized overload that takes an arbitrary functor? See for instance what I did with `_swift_withWin32DbgHelpLibrary()`. An overload means we can just directly pass a lambda with captures rather than worrying about `context`.)

Reply:
That seems like a good idea.